comp/process/expvars/expvarsimpl/expvars.go

```diff
@@ -59,6 +60,11 @@ func newExpvarServer(deps dependencies) (expvars.Component, error) {
 		return nil, err
 	}
 
+	if flavor.GetFlavor() != flavor.ProcessAgent {
+		// Don't run the server outside of the process agent
+		return nil, nil
```

Comment:
All components should be not nil. Returning `nil` here force the user of  `expvars.Component` to know that the component can be nil. 
We usually use optional.Option to make it explicit that the component is optional.